STM32F405RGTx/Core/Inc/bsp.h

```diff
@@ -3,19 +3,19 @@
 #pragma once
 
 // Contactor macros
-#define TURN_ON_CONT1_PIN() (HAL_GPIO_WritePin(CONT1_GPIO_Port, CONT1_Pin, 1))
-#define TURN_OFF_CONT1_PIN() (HAL_GPIO_WritePin(CONT1_GPIO_Port, CONT1_Pin, 0))
+#define TURN_ON_CONT1_PIN() (HAL_GPIO_WritePin(CONT1_GPIO_Port, CONT1_Pin, (GPIO_PinState)1))
+#define TURN_OFF_CONT1_PIN() (HAL_GPIO_WritePin(CONT1_GPIO_Port, CONT1_Pin, (GPIO_PinState)0))
 
-#define TURN_ON_CONT2_PIN() (HAL_GPIO_WritePin(CONT2_GPIO_Port, CONT2_Pin, 1))
-#define TURN_OFF_CONT2_PIN() (HAL_GPIO_WritePin(CONT2_GPIO_Port, CONT2_Pin, 0))
+#define TURN_ON_CONT2_PIN() (HAL_GPIO_WritePin(CONT2_GPIO_Port, CONT2_Pin, (GPIO_PinState)1))
+#define TURN_OFF_CONT2_PIN() (HAL_GPIO_WritePin(CONT2_GPIO_Port, CONT2_Pin, (GPIO_PinState)0))
 
 // Precharger macros
-#define TURN_ON_PRECHARGE_PIN() (HAL_GPIO_WritePin(PRECHARGE_GPIO_Port, PRECHARGE_Pin, 1))
-#define TURN_OFF_PRECHARGE_PIN() (HAL_GPIO_WritePin(PRECHARGE_GPIO_Port, PRECHARGE_Pin, 0))
+#define TURN_ON_PRECHARGE_PIN() (HAL_GPIO_WritePin(PRECHARGE_GPIO_Port, PRECHARGE_Pin, (GPIO_PinState)1))
+#define TURN_OFF_PRECHARGE_PIN() (HAL_GPIO_WritePin(PRECHARGE_GPIO_Port, PRECHARGE_Pin, (GPIO_PinState)0))
 
 // Debug pin macros
-#define TURN_ON_DEBUG_PIN() (HAL_GPIO_WritePin(DEBUG_GPIO_Port, DEBUG_Pin, 1))
-#define TURN_OFF_DEBUG_PIN() (HAL_GPIO_WritePin(DEBUG_GPIO_Port, DEBUG_Pin, 0))
+#define TURN_ON_DEBUG_PIN() (HAL_GPIO_WritePin(DEBUG_GPIO_Port, DEBUG_Pin, (GPIO_PinState)1))
```

Comment:
Turning on and off the debug pin needs to be as fast as possible in order to get accurate timing measurements. As such, we should be doing this at a register level:

e.g., to turn it on:

```C
DEBUG_GPIO_Port |= Debug_Pin;
```